bin.src/make_remote_butler.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+# This file is part of prompt_prototype.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+
+"""Simple script for creating a repository at a remote URI, given
+a source repository and export file.
+
+For most values of --target-repo and --seed-config, this script is only useful
+if run from the prompt-proto project on Google Cloud.
+
+The user is responsible for clearing any old copies of the repository from
+both the target URI and the registry database.
+"""
+
+
+import argparse
+import logging
+import os
+import sys
+
+import lsst.log
+from lsst.utils.timer import time_this
+from lsst.daf.butler import Butler
+
+
+def _make_parser():
+    parser = argparse.ArgumentParser()
+    # Could reasonably be positional arguments, but keep them as keywords to
+    # prevent users from confusing --src-repo with --target-repo.
+    parser.add_argument("--src-repo", required=True,
+                        help="The location of the repository whose files are to be copied.")
+    parser.add_argument("--target-repo", required=True,
+                        help="The URI of the repository to create.")
+    parser.add_argument("--seed-config",
+                        default=os.path.join(os.environ["PROMPT_PROTOTYPE_DIR"], "etc", "db_butler.yaml"),
+                        help="The config file to use for the new repository. Defaults to etc/db_butler.yaml.")
+    parser.add_argument("--export-file", default="export.yaml",
+                        help="The export file containing the repository contents. Defaults to ./export.yaml.")
+    return parser
+
+
+def main():
+    logging.basicConfig(level=logging.INFO, stream=sys.stdout)
+    lsst.log.configure_pylog_MDC("DEBUG", MDC_class=None)
+
+    args = _make_parser().parse_args()
+    seed_config = lsst.daf.butler.Config(args.seed_config)
+    logging.info("Creating repository at %s...", args.target_repo)
```

Comment:
Why does the make_local_butler.py file exist? Isn't that identical to this version if the default value for `--target-repo` was "."?  Have you considered making this command a butler pluggable command so that you could make use of the normal `butler --help` click infrastructure? (it's butler create + butler import).

Reply:
I do not believe that integrating what should be a local, highly specific script into the `butler` ecosystem would be beneficial, even if we had the time to do so.

We are deliberately not using the `butler` command-line itself because we need to understand what our overheads are, and the `butler` utility loses a lot of time on package import and `Butler` setup.